chore(setup): `lerna` for monorepo package versioning

xamdoo opened the pull request:
<!-- ✨ Thanks for making a pull request! -->

### *What* change does this PR introduce?

This PR sets up Lerna for managing multiple packages within a single repository
### *Why* was this change needed?
Lerna would handle shared dependencies among packages within the monorepo reducing duplication and simplifying installation
<!-- 🚫 If your PR closes an open issue, link your PR to the issue with `Closes #16` or `Fixes #16`, where #16 is the issue number. For example:
- Closes #16
- Fixes #42
Please put the close or fix statement in a bullet point so that it renders the issue title as well. -->

- Closes #48 

### Other information (screenshots)

<!-- 📸 If you have any, please provide screenshots that would give additional visual context. Other forms of information (links to resources) would also be welcome. -->

### Checklist

<!-- 📋 Here's a checklist of what needs to be done before this PR is ready for review. Once you have completed this checklist, you may request a review, and we will get back to you shortly. -->

- [X] 📝 I have reviewed my code prior to submission
- [X] ❔ Issue number has been linked (if applicable)
- [ ] 🧪 Tests have been added or updated (if applicable)
- [ ] 📖 Documentation has been updated (if applicable)
- [ ] 📸 Screenshots added (if applicable)

xamdoo committed:
chore(*): setup lerna

RickyC0626 commented on lerna.json:
```diff
@@ -0,0 +1,4 @@
+{
+  "$schema": "node_modules/lerna/schemas/lerna-schema.json",
+  "version": "0.0.0"
```
**chore:** Let's bump to version `0.0.1`

RickyC0626 reviewed (changes requested):
Remember to update the working directories in GitHub Actions workflows too.

RickyC0626 commented on packages/backend/package.json:
**chore:** Set version to `0.0.1` here as well.

xamdoo replied:
does this apply to both frontend and backend packages or just the backend?

RickyC0626 replied:
Oh right, both. I think an easy way to do this is use the `lerna version` command with the flags `--no-changelog` and `--no-git-tag-version`. We can run the same command without flags during actual release.

xamdoo committed:
chore(deps): bump version to v0.0.1

RickyC0626 reviewed (dismissed):
Looks good!

jaysabe reviewed (changes requested):
Looks like we'll need to address merging conflicts before proceeding with the merge to dev.

RickyC0626 commented:
Since we have new updates to the original files, we probably have to move the directories to `packages/` again. `git mv` will be helpful for those. Perhaps 1 commit for each directory, as separate PRs. After renaming we can apply the `lerna` changes on top.

RickyC0626 committed:
refactor(ci): set working directory per step

Co-Authored-By: xamdoo <75206869+xamdoo@users.noreply.github.com>

RickyC0626 committed:
chore(deps): add husky dependency for each package

This is necessary for lerna to work, as it handles dependencies per
package/workspace

Co-Authored-By: xamdoo <75206869+xamdoo@users.noreply.github.com>

CeeJayyy007 reviewed (approved):
**praise:** Awesome work!
